### ISSUE TITLE
Calcite Tree Item - "ARIA toggle fields must have an accessible name"

### ISSUE THREAD

**Issue by abp6318** [Esri/calcite-design-system#8048]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

Using aXe DevTools, Hub Discovery realized that the underlying issue for inaccessible names on calcite tree items is from within the calcite component itself. We are requesting an update to this, and expect it to be a small change as we have a list of options that also use checkboxes in a similar manner, and the only difference is a single `div` wrapping the checkbox and it's option label together within the overarching label.

(I don't see any issues related to this; apologies if I overlooked one)

### Actual Behavior

![Screenshot 2023-10-24 at 11 39 13 AM](https://github.com/Esri/calcite-design-system/assets/60486980/4df283d9-6130-42d8-819e-6937f5d761cc)
![Screenshot 2023-10-24 at 11 39 46 AM](https://github.com/Esri/calcite-design-system/assets/60486980/a023199a-c5cc-4977-9ea6-df053d077d35)

----
calcite-tree-item's implementation causes this (even in code pens):
```javascript
<label class={CSS.checkboxLabel} key="checkbox-label">
<!-- missing div here -->
          <calcite-checkbox
            checked={this.selected}
            class={CSS.checkbox}
            data-test-id="checkbox"
            indeterminate={this.hasChildren && this.indeterminate}
            scale={this.scale}
            tabIndex={-1}
          />
          {defaultSlotNode}
<!-- missing /div here -->
</label>
```

![image](https://github.com/Esri/calcite-design-system/assets/60486980/58bf308a-481e-4423-b81d-0731804fa9f3)

-----

Our implementation of arcgis-hub-facet-options is this:

```

return <calcite-label key={key}>
        <div class="option-container"> <!-- this is what we need to get added in calcite (I think) -->
          <calcite-checkbox
            checked={option.selected}
            id={option.key}
            name={option.label}
            onCalciteCheckboxChange={checkboxChangeHandler}
          />
          <span class="option">{option.label}</span>
        </div> 
</calcite-label>

```

results in:
![Screenshot 2023-10-24 at 11 41 59 AM](https://github.com/Esri/calcite-design-system/assets/60486980/dcbf581a-6e97-47a6-b2ab-7f977e801886)

-----

I've been trying to find a temporary fix but I think the only way to temp fix this would also be through calcite.
- [The component](https://github.com/Esri/calcite-design-system/blob/4a8106fb4726ab65778f8db2ca52d1133cc8835f/packages/calcite-components/src/components/tree-item/tree-item.tsx#L227-L237)
- [using this checkbox, that has an aria-label](https://github.com/Esri/calcite-design-system/blob/4a8106fb4726ab65778f8db2ca52d1133cc8835f/packages/calcite-components/src/components/checkbox/checkbox.tsx#L262)
- [defaulting to the component label](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/utils/label.ts#L170), rather than the content that we would rather use.

-----

Please see code snippet above for idea on how to further implement this.

### Expected Behavior

Our implementation of arcgis-hub-facet-options is this:

```javascript
return <calcite-label key={key}>
        <div class="option-container"> <!-- this is what we need to get added in calcite (I think) -->
          <calcite-checkbox
            checked={option.selected}
            id={option.key}
            name={option.label}
            onCalciteCheckboxChange={checkboxChangeHandler}
          />
          <span class="option">{option.label}</span>
        </div> 
</calcite-label>
```

results in:
![Screenshot 2023-10-24 at 11 41 59 AM](https://github.com/Esri/calcite-design-system/assets/60486980/dcbf581a-6e97-47a6-b2ab-7f977e801886)

----

We want something similar for calcite-tree-item

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/tree/

### Reproduction Steps

1. Open aXe DevTools
2. Run page scan
3. Look for "ARIA toggle fields must have an accessible name" (labeled as a serious issue)

### Reproduction Version

latest

### Relevant Info

Issue appears in Chrome and Safari (minimum).

### Priority impact

p2 - want for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

@thomas-hervey pinging for eyes

**Comment by geospatialem** [comment]
@abp6318 It looks like in the first and last screenshots the issue pertains to when `tree` is in `selectionMode` of `"ancestors"`, does that seem accurate?

If so, we do have an open issue from the 2022 audit, targeted for a fix next month (November) with #5615.

**Comment by abp6318** [comment]
Yep! Sorry @geospatialem geospatialem for the duplicate issue! Thank you for looking into it so quickly. Much appreciated.

**Comment by abp6318** [comment]
I did a good amount of digging yesterday and provided what I believe is a possible solution. If you think it wise, could we attach my notes to that ticket @geospatialem?

**Comment by geospatialem** [comment]
> Thank you for looking into it so quickly. Much appreciated.

Appreciate the issue filing! 

> I did a good amount of digging yesterday and provided what I believe is a possible solution. If you think it wise, could we attach my notes to that ticket @geospatialem?

That'd be excellent! Could you add the context in to the issue? Will definitely speed up the process of fixing in the code. 🙌🏻 

Once you add it in, we can close out the above issue in favor of the audit issue.

**Comment by abp6318** [comment]
@geospatialem Comment made! -- https://github.com/Esri/calcite-design-system/issues/5615#issuecomment-1777594837

Thanks for the help with this.

**Comment by geospatialem** [comment]
Closing out the above in favor of #5615.
